atomicapp/providers/kubernetes.py

```diff
@@ -204,3 +205,52 @@ def undeploy(self):
             if self.config_file:
                 cmd.append("--kubeconfig=%s" % self.config_file)
             self._call(cmd)
+
+    def persistent_storage(self, graph, action):
+        """
+        All extra methods that belong to 'requirements'
+        Each method will be passed the respective graph from Nulecule
+        and any other extra information that is required.
+
+        Actions are either: run, stop or uninstall as per the Requirements class
+
+        Curently run is the only function implemented for k8s persistent storage
+        """
+
+        logger.debug("Persistent storage enabled! Running action: %s" % action)
+
+        self._check_persistent_volumes()
+
+        # Get the path of the persistent storage yaml file includes in /external
+        # Plug the information from the graph into the persistent storage file
+        base_path = os.path.dirname(os.path.realpath(__file__))
+        template_path = os.path.join(base_path, 'external/kubernetes/persistent_storage.yaml')
+        with open(template_path, 'r') as f:
+            content = f.read()
+            template = Template(content)
+            rendered_template = template.safe_substitute(graph)
+
+        # Must write persistent storage to a tmp file as to not have malicious intent
+        # when running direct cmd
```

Comment:
Change this comment.. As we discussed on friday I don't think there is any chance of malicious intent.. Maybe just change it to "Write out to temporary file to pass to kubectl"